src/mqtt_client/AtlasMqttClient.cpp

```diff
@@ -0,0 +1,78 @@
+#include "AtlasMqttClient.h"
+#include "AtlasMqttClient_callback.h"
+#include "AtlasMqttClient_actionListener.h"
+#include "AtlasMqttLogger.h"
+#include "AtlasMqttException.h"
+#include <mqtt/async_client.h>
+
+namespace Atlas
+{
+AtlasMqttClient* AtlasMqttClient::instance = nullptr;
+
+AtlasMqttClient& AtlasMqttClient::getInstance()
+{
+    if (instance == nullptr)
+    {
+        instance = new AtlasMqttClient;
+    }
+    return *instance;
+}
+
+void AtlasMqttClient::destroyInstance()
+{
+    if (instance != nullptr)
+    {
+        delete instance;
+        instance = nullptr;
+    }
+}
+
+void AtlasMqttClient::publishMessage(const string address, const string clientID, const string topic, 
+                    const string message, int QoS)
+{
+    AtlasMqttLogger::getInstance().logMessage("Initializing client (" + clientID + 
+                        ") connection to server (" + address + ")");
+
+    mqtt::async_client client(address, clientID);
+    AtlasMqttClient_callback cb;
+    client.set_callback(cb);
+    mqtt::connect_options connOps;
+    
+    try
+    {
+        mqtt::token_ptr connTok = client.connect(connOps);
+        connTok->wait();
```

Comment:
I guess this operation is blocking, right?

Reply:
Yes, but it gives a rapid validation of an action.